core/assets/CMakeLists.txt

```diff
@@ -0,0 +1,120 @@
+#
+# Copyright Soramitsu Co., Ltd. All Rights Reserved.
+# SPDX-License-Identifier: Apache-2.0
+#
+
+add_library(assets
+    assets.cpp
+    )
+
+find_program(ECHO echo)
+
+if(EMBEDDINGS)
+    set(SOURCE ${PROJECT_SOURCE_DIR}/examples/first_kagome_chain/localchain.json)
+    if(NOT EXISTS ${SOURCE})
+        fatal_error("Not found resource for embedded assets: ${SOURCE}")
+    endif()
+    set(OUTPUT embedded_chainspec.hpp)
+    add_custom_command(
```

Comment:
Maybe rewrite with https://cmake.org/cmake/help/latest/command/configure_file.html?
Should be shorter.

Reply:
This feature uses to use cmake options and definition (their values) in some .h/.hpp files. 
In our case is used big raw string content. And for keys we need to scan directory. 
As result `configure_file` is inaproptiate here.